File: dsa/helltime1.0/backtracking/78.cpp
class Solution
{
public:
    vector<vector<int>> subsets(vector<int> &nums)
    {
        vector<vector<int>> answer;
        vector<int> combination;
        function<void(int)> dfs = [&](int index) -> void
        {
            if (index == nums.size())
            {
                answer.push_back(combination);
                return;
            }
            dfs(index + 1);
            combination.push_back(nums[index]);
            dfs(index + 1);
            combination.pop_back();
        };
        dfs(0);
        return answer;
    }
};

File: dsa/helltime1.0/extras2.0/78.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        out = []

        comb = 1 << len(nums)

        for mask in range(comb):
            subset = []
            for i in range(len(nums)):
                if mask & (1 << i):
                    subset.append(nums[i])
            out.append(subset)

        return out